packages/e2e-tests/specs/performance/performance.test.js

```diff
@@ -47,22 +47,39 @@ function isKeyUpEvent( item ) {
 	return isKeyEvent( item ) && item.args.data.type === 'keyup';
 }
 
+function isFocusEvent( item ) {
+	return isKeyEvent( item ) && item.args.data.type === 'focus';
+}
+
 function getEventDurationsForType( trace, filterFunction ) {
 	return trace.traceEvents
 		.filter( filterFunction )
 		.map( ( item ) => item.dur / 1000 );
 }
 
-function getEventDurations( trace ) {
+function getTypingEventDurations( trace ) {
 	return [
 		getEventDurationsForType( trace, isKeyDownEvent ),
 		getEventDurationsForType( trace, isKeyPressEvent ),
 		getEventDurationsForType( trace, isKeyUpEvent ),
 	];
 }
 
+function getSelectionEventDurations( trace ) {
+	return [ getEventDurationsForType( trace, isFocusEvent ) ];
+}
+
+jest.setTimeout( 1000000 );
```

Comment:
@youknowriad what was the rationale behind setting a default timeout of 16 minutes? I was hoping to find some explanatory comment or git message or PR comment but I didn't see any. is this something we might want to revise?

Reply:
It's just a random number because jest has a short timeout by default causing the job to actually fail. It's basically the "maximum time one job run can take"

Reply:
Note that previously all metrics were in the same test (right now we splitted them)

Reply:
So far, I haven't seen any test fail because of this personally in CI. Did you see any previous cases where this would have helped? I don't mind changing the number personally but I wanted to avoid cases where it fails but in reality it shouldn't have, it was just taking a bit longer

Reply:
Maybe you are right: I thought I had seen a number of failures from these timeouts (and maybe I did, but I can't find them - I've run around three hundred runs of the performance tests on empty changes that shouldn't fail).

There are some example failures [here](https://github.com/WordPress/gutenberg/actions/runs/3238266544/jobs/5377308452) and [here](https://github.com/WordPress/gutenberg/actions/runs/3237439063/attempts/9), and [here](https://github.com/WordPress/gutenberg/actions/runs/3295158389/jobs/5462668918), but these seem to be timing out after around a minute (entire test run fails around an hour) - I'm not sure where that timeout comes from since it's not sixteen minutes and I don't see any overrides in the site-editor perf test suite that correspond to the timeouts seemingly caused by waiting for `toggleGlobalBlockInserter()`.

Then again, maybe this is exactly what we're seeing, because the total test is failing at some point from the sixteen minute timeout; I'm really not sure. The "Measuring loading time." loop allows for more than twelve minutes of waiting time _before_ we get to the timeout failure, so it's not hard to guess that this is what's happening, though I don't see clear evidence indicating one way or another (need to probe failing flakey tests).

---

Back to the point I might ask the same question in reverse: are there legitimate test runs where waiting for sixteen minutes before giving up helps? If we've ever gotten to five minutes (based on the history of test runs) then any further delay is _only_ going to let a failed test take longer to conclude; there is no situation in which extending that delay is helpful, no situation with a successful test, or with any false positive or negative.

It does seem that in the tests I've run the block inserter is in the error messages but I'm not sure that's the problem; it could be the unlucky component at the coincidental juncture of bad test runs.